vanilla-tests/src/main/java/fixtures/streamstyleserialization/models/Basic.java

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package fixtures.streamstyleserialization.models;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.core.util.serializer.JsonUtils;
+import com.azure.json.JsonReader;
+import com.azure.json.JsonSerializable;
+import com.azure.json.JsonToken;
+import com.azure.json.JsonWriter;
+
+/** The Basic model. */
+@Fluent
+public final class Basic implements JsonSerializable<Basic> {
+    private Integer id;
+
+    private String name;
+
+    private CMYKColors color;
+
+    /**
+     * Get the id property: Basic Id.
+     *
+     * @return the id value.
+     */
+    public Integer getId() {
+        return this.id;
+    }
+
+    /**
+     * Set the id property: Basic Id.
+     *
+     * @param id the id value to set.
+     * @return the Basic object itself.
+     */
+    public Basic setId(Integer id) {
+        this.id = id;
+        return this;
+    }
+
+    /**
+     * Get the name property: Name property with a very long description that does not fit on a single line and a line
+     * break.
+     *
+     * @return the name value.
+     */
+    public String getName() {
+        return this.name;
+    }
+
+    /**
+     * Set the name property: Name property with a very long description that does not fit on a single line and a line
+     * break.
+     *
+     * @param name the name value to set.
+     * @return the Basic object itself.
+     */
+    public Basic setName(String name) {
+        this.name = name;
+        return this;
+    }
+
+    /**
+     * Get the color property: The color property.
+     *
+     * @return the color value.
+     */
+    public CMYKColors getColor() {
+        return this.color;
+    }
+
+    /**
+     * Set the color property: The color property.
+     *
+     * @param color the color value to set.
+     * @return the Basic object itself.
+     */
+    public Basic setColor(CMYKColors color) {
+        this.color = color;
+        return this;
+    }
+
+    /**
+     * Validates the instance.
+     *
+     * @throws IllegalArgumentException thrown if the instance is not valid.
+     */
+    public void validate() {}
+
+    @Override
+    public JsonWriter toJson(JsonWriter jsonWriter) {
+        return jsonWriter.flush();
+    }
+
+    public static Basic fromJson(JsonReader jsonReader) {
+        return JsonUtils.readObject(
+                jsonReader,
+                reader -> {
+                    Integer id = null;
+                    String name = null;
+                    CMYKColors color = null;
+                    while (reader.nextToken() != JsonToken.END_OBJECT) {
+                        String fieldName = reader.getFieldName();
+                        reader.nextToken();
+
+                        if ("id".equals(fieldName)) {
+                            if (reader.currentToken() != JsonToken.NULL) {
```

Comment:
[nit] would be great to not repeat `reader.currentToken() != JsonToken.NULL` on every primitive type, maybe reader can have an overload to read nullable result or some util methods can help?

Perhaps it's ok with generated code, but easy to forget when writing manually

Reply:
I think updating the JsonReader is the right choice here as this would be a pattern direct users of JsonReader would need to copy as well